src/sinks/pulsar/tests.rs

```diff
@@ -0,0 +1,116 @@
+#[cfg(test)]
+mod config {
+    use crate::sinks::pulsar::config::PulsarSinkConfig;
+    use vector_config::component::GenerateConfig;
+
+    #[test]
+    fn generate_config() {
+        PulsarSinkConfig::generate_config();
+    }
+}
```

Comment:
I think the extra module here for config isn't needed. Instead, let's just have a file `tests.rs` for the unit tests, and a file `integration_tests.rs` for the integration tests, for consistency.